src/plugins/dashboard/public/application/listing/dashboard_listing.tsx

```diff
@@ -62,11 +63,21 @@ export const DashboardListing = ({
       chrome: { setBreadcrumbs },
     },
   } = useKibana<DashboardAppServices>();
-
   const [unsavedDashboardIds, setUnsavedDashboardIds] = useState<string[]>(
     dashboardSessionStorage.getDashboardIdsWithUnsavedChanges()
   );
 
+  useEffect(() => {
+    customEvents.setCustomEventContext({
+      page: 'list',
+    });
+    return () => {
+      customEvents.setCustomEventContext({
+        page: undefined,
+      });
+    };
+  }, []);
+
```

Comment:
Idea: We could extend https://github.com/elastic/kibana/blob/main/src/plugins/usage_collection/public/components/track_application_view/track_application_view_component.tsx to automatically set event context :)

Or provide an API that does this to avoid code repetition on every view. What do you think?

Reply:
I would make it a hook, since it can be used in very different ways per app.
I know that ` TrackApplicationViewComponent` can be coexist on screen (for example in the flyout), while a page is a singular entity I believe.

Reply:
Made a hook out of it :-)

Reply:
> I know that two ` TrackApplicationViewComponent`s can be coexist on screen (for example in the flyout), while a page is a singular entity I believe.

That is a very valid point! Which, IMO, it highlights a potential issue when sharing the context in the UI. I'm thinking about tracking general user clicks. ~~When two `sub-views` coexist, it'll be hard to identify which one it happened in.~~

Scratch that! The component could actually emit an event "click" and provide the sub-view info in the additional event's params (not the context). All good here 👍